### PR TITLE
Allow clipping layout maps to non-rectangular sizes

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows users to clip a map item to a shape or polygon item from their layout, allowing for non-rectangular maps in the layout

Peek 2020-07-29 15-39

### CODE REVIEW

File: src/ui/layout/qgslayoutmapclippingwidgetbase.ui [hunk 1]
```diff
@@ -98,6 +98,51 @@
...
+        <property name="text">
+         <string>The clipping mode determines how features from vector layers will be clipped.</string>
+        </property>
+        <property name="wordWrap">
+         <bool>true</bool>
+        </property>
+       </widget>
+      </item>
+      <item row="0" column="0">
+       <widget class="QLabel" name="mClipToAtlasLabel_2">
```
Comment (nirvn) [inline src/ui/layout/qgslayoutmapclippingwidgetbase.ui:121]:
```suggestion
       <widget class="QLabel" name="mClipToItemLabel">
```

**Review by nirvn — COMMENTED** [review]
Looking good. Only one tiny .ui change highlighted.

**Comment by GSavelev-VSEGEI** [comment]
This feature for good needs to be improved by clipping grids as well (as it is in arcmap)

it will be useful for making maps like this:
image

is a new feature request required?
how do I sposor it if it is needed?

**Comment by gioman** [comment]
> is a new feature request required?

yes thanks, but there is not guarantee that will be implemented soon or at all.



>  how do I sponsor it if it is needed?

if you need the feature sooner than later, hire a developer:

https://qgis.org/en/site/forusers/commercial_support.html

**Pull request merged**